cmake/sdlplatform.cmake

```diff
@@ -42,6 +42,8 @@ macro(SDL_DetectCMakePlatform)
         set(SDL_CMAKE_PLATFORM IOS)
       endif()
     endif()
+  elseif(QNX)
```

Comment:
This does not detect QNX properly. First, `QNX` is not yet defined. Second, it should be under the `UNIX` section above.

```
diff --git a/cmake/sdlplatform.cmake b/cmake/sdlplatform.cmake
index 536a3fb2b..6d8ece6ac 100644
--- a/cmake/sdlplatform.cmake
+++ b/cmake/sdlplatform.cmake
@@ -28,6 +28,8 @@ macro(SDL_DetectCMakePlatform)
       set(SDL_CMAKE_PLATFORM AIX)
     elseif(CMAKE_SYSTEM_NAME MATCHES "Minix.*")
       set(SDL_CMAKE_PLATFORM MINIX)
+    elseif(CMAKE_SYSTEM_NAME MATCHES "QNX")
+      set(SDL_CMAKE_PLATFORM QNX)
     endif()
   elseif(APPLE)
     if(CMAKE_SYSTEM_NAME MATCHES ".*Darwin.*")
@@ -42,8 +44,6 @@ macro(SDL_DetectCMakePlatform)
         set(SDL_CMAKE_PLATFORM IOS)
       endif()
     endif()
-  elseif(QNX)
-    set(SDL_CMAKE_PLATFORM QNX)
   elseif(CMAKE_SYSTEM_NAME MATCHES "BeOS.*")
     message_error("BeOS support has been removed as of SDL 2.0.2.")
   elseif(CMAKE_SYSTEM_NAME MATCHES "Haiku.*")
```